### DISCUSSION TITLE
Bitfocus starts before Resolume causing it to fail to be recognized as connected.

### DISCUSSION THREAD

**Discussion by justinmquinn** [bitfocus/companion#3173] [General]
We have a pretty simple set up with Resolume Arena and Bitfocus companion both running on the same Mac. Of course, the objective is to use the stream deck to control Resolume, which works quite well. However, the problem that we have is- when the machine first starts up bit focus starts immediately and Resolume starts shortly afterwards, which causes Companion to not see the Resolume instance as online. Our team has to manually go into the UI and uncheck and recheck the active toggle next to the Resolume instance. We are using this in a church setting, so this is an inconvenient step that we have to do frequently.

Is there any setting in companion that causes it to constantly re-scan for devices coming online?

**Comment by dnmeid** [comment]
First of all you should contact the maintainer of the resolume module. Normally a module that can't connect should retry the connection periodically. This should be fixed at the module level.

As a workaround for now you can automate switch off / switch on with a trigger. There is a trigger event "Startup". Give it a delay of how long Resolume usually needs longer. In the actions disable Resolume and then enable it again with maybe another 500ms delay. You can also give the trigger a condition that the Resolume connection status needs to be not good (= status OK + Invert on) to run the trigger, so it will not do the disable/enable if Resolume is already connected when you start Companion.
